src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -82,29 +80,26 @@ case class DepositIngestTask(deposit: Deposit, dansBagValidator: DansBagValidato
     // TODO: delete draft if something went wrong
   }
 
-  private def formatViolation(v: (String, String)): String = v match {
-    case (nr, msg) => s" - [$nr] $msg"
+  private def getPersistentId(response: DataverseResponse[DatasetCreationResult]): Try[String] = Try {
+    response.data.get.persistentId.toString
+  }
+
+  private def extractDatasetVersion(json: String): Try[DatasetVersion] = Try {
+    (JsonMethods.parse(json) \ "datasetVersion").extract[DatasetVersion]
   }
 
   private def uploadFilesToDataset(datasetId: String): Try[Unit] = {
     trace(datasetId)
     for {
       filesXml <- deposit.tryFilesXml
       ddm <- deposit.tryDdm
-      defaultRestrict <- Try { (ddm \ "profile" \ "accessRights").headOption.forall(AccessRights toDefaultRestrict) }
+      defaultRestrict = (ddm \ "profile" \ "accessRights").headOption.forall(AccessRights toDefaultRestrict)
```

Comment:
Why this change?

Reply:
Joke told me that this is a more concise way of doing the same. I may have misunderstood.

Reply:
No, it seems you (and @jo-pol) are right. I was not sure if an exception raised on the right-hand side of the `=` would be handled the same as in a `<- Try { ...}` so I did a little test in a scala workbook:
```scala
import scala.util.Try

// With  <- Try {}
for {
  _ <- Try { throw new RuntimeException("test")}
} yield()


// With =
try
{
  for {
    _ <- Try { } // Have to put this here, otherwise it does not compile
    _ = throw new RuntimeException("test")
  } yield ()
}
catch {
  case e: Exception => println("caught")
}

```
Both return a `Failure` and no "caught" is printed, so that proves your point.